src/main/java/com/kichik/pecoff4j/util/Strings.java

```diff
@@ -1,15 +1,9 @@
 package com.kichik.pecoff4j.util;
 
-import java.io.UnsupportedEncodingException;
-
 public class Strings {
 
 	public static int getUtf16Length(String string) {
-		try {
-			return string.getBytes("UTF-16").length + 2; // null terminator
-		} catch (UnsupportedEncodingException e) {
-			return string.length() * 2 + 2;
-		}
+		return string.length() * 2 + 2; // null terminator
```

Comment:
I would rather not use `string.length() * 2` as it's not going to always be correct. Some characters can translate to 4 bytes. Maybe we can try `UTF-16LE` so there is no BOM? https://stackoverflow.com/a/2531819/492773

Reply:
Thank you for your quick response. I was relying on to the JavaDoc that length() returns the "number of Unicode code units in the string". Code units have a fixed length (see e.g. https://developer.mozilla.org/en-US/docs/Glossary/Code_unit), which is 16 bit for Java. So `"😀".length()` is still 2 and not 1, which is exactly what we want.
Of course, your proposal would fix the problem as well (while still requiring the try-catch). Whatever you prefer...

Reply:
TIL. Thanks for looking into it.